### PR TITLE
Changed the way of storing flow session

### PR DESCRIPTION
**Pull request by rafael-santiago-take**
Changed the way of storing flow session by transfering from StateSessionManager to FlowSessionManager

### CODE REVIEW

File: src/Take.Blip.Builder/FlowManager.cs [hunk 1]
```diff
@@ -84,8 +84,8 @@ IFlowLoader flowLoader
             _userOwnerResolver = userOwnerResolver;
             _inputExpirationHandler = inputExpirationHandler;
             _applicationNode = application.Node;
-            _stateSessionManager = stateSessionManager;
             _flowLoader = flowLoader;
+            this._flowSessionManager = flowSessionManager;
```
Comment (caiopsouza) [inline src/Take.Blip.Builder/FlowManager.cs:88]: You don't need `this` here.

File: src/Take.Blip.Builder/IFlowSessionManager.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace Take.Blip.Builder
+{
+
```
Comment (caiopsouza) [inline src/Take.Blip.Builder/IFlowSessionManager.cs:6]: Remove this line.

File: src/Take.Blip.Builder/IFlowSessionManager.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    /// </summary>
+    public interface IFlowSessionManager
+    {
+        /// <summary>
+        /// Gets the current flow session for the user in the flow.
+        /// </summary>
+        Task<string> GetFlowSessionAsync(IContext context, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Gets the current flow session for the user in the flow.
```
Comment (caiopsouza) [inline src/Take.Blip.Builder/IFlowSessionManager.cs:18]: Should this be `sets`?

File: src/Take.Blip.Builder/FlowSessionManager.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        /// <summary>
+        /// Gets the current flow session for the user in the flow.
+        /// </summary>
+        public Task<string> GetFlowSessionAsync(IContext context, CancellationToken cancellationToken)
+        {
+            return context.GetContextVariableAsync(CURRENT_FLOW_SESSION_KEY, cancellationToken);
+        }
+
+        /// <summary>
+        /// Gets the current flow session for the user in the flow.
```
Comment (caiopsouza) [inline src/Take.Blip.Builder/FlowSessionManager.cs:22]: Should this be `sets`?

File: src/Take.Blip.Builder/FlowSessionManager.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    /// <summary>
+    /// Defines a service for managing the current flow session for a given user.
+    /// </summary>
+    public class FlowSessionManager : IFlowSessionManager
+    {
+        private const string CURRENT_FLOW_SESSION_KEY = "currentFlowSession";
+
+        /// <summary>
+        /// Gets the current flow session for the user in the flow.
+        /// </summary>
```
Comment (fadoaglauss) [inline src/Take.Blip.Builder/FlowSessionManager.cs:15]: You don't need to add xml documentation in the implementation files, just in the interface files.

**Review by fadoaglauss — APPROVED** [review]

File: src/Take.Blip.Builder/FlowSessionManager.cs (hunk 5 above)
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/FlowSessionManager.cs:15]: currently not adding xml documentation in the implementation file, generates warnings

File: src/Take.Blip.Builder/FlowSessionManager.cs (hunk 4 above)
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/FlowSessionManager.cs]: done

File: src/Take.Blip.Builder/IFlowSessionManager.cs (hunk 3 above)
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/IFlowSessionManager.cs]: done

File: src/Take.Blip.Builder/FlowManager.cs (hunk 1 above)
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/FlowManager.cs]: done

File: src/Take.Blip.Builder/IFlowSessionManager.cs (hunk 2 above)
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/IFlowSessionManager.cs]: done

**Review by caiopsouza — APPROVED** [review]

File: src/Take.Blip.Builder/FlowManager.cs [hunk 6]
```diff
@@ -409,7 +409,7 @@ private async Task<(Flow, State, StateTrace, Stopwatch)> RedirectToSubflowAsync(
             context.Flow = subflow;
             var newState = subflow.States.Single(s => s.Root);
 
-            await _stateSessionManager.SetStateAsync(userIdentity, shortNameOfSubflow, cancellationToken);
```
Comment (jujucasseb) [inline src/Take.Blip.Builder/FlowManager.cs:412]: Is this method still being used? If not remove its implementation

**Review by jujucasseb — APPROVED** [review]

File: src/Take.Blip.Builder/FlowManager.cs (hunk 6 above)
Comment (rafael-santiago-take) [reply src/Take.Blip.Builder/FlowManager.cs:412]: It's was a ethod that already existed...

**Pull request merged**